### PR TITLE
Add arbiter count to client volfile

### PR DESCRIPTION
**Pull request by vatsa287**
Add arbiter-count to client volfile under afr xlator

Signed-off-by: Shree Vatsa N <vatsa@kadalu.tech>

### CODE REVIEW

**Comment by vatsa287** [comment]
@aravindavk Is above PR correct or should it be,
```
    {% if dist_grp.type == "arbiter" %}
      option arbiter-count 1
    {% endif %}
```

**Comment by aravindavk** [comment]
> @aravindavk Is above PR correct or should it be,
> 
> ```
>     {% if dist_grp.type == "arbiter" %}
>       option arbiter-count 1
>     {% endif %}
> ```

count based one looks correct.

**Comment by aravindavk** [comment]
Add `arbiter_count` here also https://github.com/kadalu/volgen/blob/main/src/volgen.cr#L59

**Comment by aravindavk** [comment]
And add test here https://github.com/kadalu/volgen/blob/main/spec/replicate_spec.cr

File: templates/client.vol.j2 [hunk 1]
```diff
@@ -42,6 +42,9 @@
     {% if dist_grp.redundancy_count > 0 %}
       option redundancy {{ dist_grp.redundancy_count }}
     {% endif %}
+    {% if dist_grp.type == "arbiter" %}
```
Comment (aravindavk) [inline templates/client.vol.j2]: count based approach is better as used earlier.

File: spec/replicate_spec.cr [hunk 2]
```diff
@@ -9,6 +9,7 @@ DIST_REP2_VOLDATA_JSON  = File.read("samples/dist_replica2.json")
 REP3_VOLDATA_JSON       = File.read("samples/replica3.json")
 DIST_REP3_VOLDATA_JSON  = File.read("samples/dist_replica3.json")
 TIEBREAKER_VOLDATA_JSON = File.read("samples/tiebreaker.json")
+ARBITER_VOLDATA_JSON    = File.read("samples/storage_unit_arbiter.json")
```
Comment (aravindavk) [inline spec/replicate_spec.cr]: Use different sample here. Copy replica3 json and add arbiter count

File: spec/replicate_spec.cr [hunk 3]
```diff
@@ -46,5 +47,10 @@ describe Volgen do
       volfile = Volgen.generate(SHD_TMPL, DIST_REP3_VOLDATA_JSON)
       volfile.should eq File.read("spec/sample_output/shd_dist_replica3.vol")
     end
+
+    it "checks if volfile rendered for Arbiter volume" do
+      volfile = Volgen.generate(REPLICA_TMPL, ARBITER_VOLDATA_JSON)
```
Comment (aravindavk) [inline spec/replicate_spec.cr]: storage unit data is passed here.

File: spec/replicate_spec.cr [hunk 4]
```diff
@@ -46,5 +47,10 @@ describe Volgen do
...
+      volfile.should eq File.read("spec/sample_output/storage_unit_arbiter.vol")
```
Comment (aravindavk) [inline spec/replicate_spec.cr]: Compare with new vol file. Copy replica3 client volfile to arbiter client volfile and add `option arbiter-count 1`

**Review by aravindavk — COMMENTED** [review]
To test locally. Run

```
crystal spec
```

**Comment by aravindavk** [comment]
> To test locally. Run
> 
> ```
> crystal spec
> ```

Or run `make spec`

File: templates/client.vol.j2 (hunk 1 above)
Comment (vatsa287) [reply templates/client.vol.j2]: Ack

File: spec/replicate_spec.cr (hunk 2 above)
Comment (vatsa287) [reply spec/replicate_spec.cr]: Done

File: spec/replicate_spec.cr (hunk 3 above)
Comment (vatsa287) [reply spec/replicate_spec.cr]: Ack. Used Replica 3, Arbiter 1 Volume Data

File: spec/replicate_spec.cr (hunk 4 above)
Comment (vatsa287) [reply spec/replicate_spec.cr]: Comparison Fails

**Review by aravindavk — COMMENTED** [review]
Changes Looks good. Also add arbiter-count to shd template as well https://github.com/kadalu/volgen/blob/main/templates/shd.vol.j2

File: spec/replicate_spec.cr (hunk 4 above)
Comment (aravindavk) [reply spec/replicate_spec.cr]: Check the diff. May be mismatch in the order of placing the option

File: spec/sample_output/replica3_arbiter.vol [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+
+volume vol1-replicate-0
+    type cluster/replicate
+    option data-self-heal on
+    option granular-entry-heal on
+    option iam-self-heal-daemon off
+    option metadata-self-heal on
+    option entry-self-heal on
+    option read-hash-mode 5
+    option arbiter-count 1
```
Comment (aravindavk) [inline spec/sample_output/replica3_arbiter.vol]: In template, this is added before `data-self-heal` option

Comment (vatsa287) [reply spec/sample_output/replica3_arbiter.vol]: Moved arbiter-count abover data-self-heal option in sample_output as it is in template

File: samples/replica3_arbiter.json [hunk 6]
```diff
@@ -0,0 +1,33 @@
+{
+    "name": "vol1",
+    "id": "e79048bf-ac4d-47cf-8d1f-0a07366ff1ea",
+    "distribute_groups": [
+       {
+           "replica_count": 3,
+           "ariter_count": 1,
```
Comment (aravindavk) [inline samples/replica3_arbiter.json]: typo

Comment (vatsa287) [reply samples/replica3_arbiter.json]: Ohh sorry my bad!!

**Review by aravindavk — APPROVED** [review]

**Pull request merged**